src/boot/efi/devicetree.c

```diff
@@ -5,6 +5,7 @@
 #include "util.h"
 
 #define FDT_V1_SIZE (7*4)
+#define be32toh(x) __builtin_bswap32(x)
```

Comment:
just for paranoia, can you add change this to:

```c
#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
#define be32toh(x) __builtin_bswap32(x)
#else
#error "Unexpected by order in EFI mode?"
#endif
```

this is formalized documentation if you so will, explaining a bit why we can just get away with defining be32toh like this

Reply:
efi.h already asserts little endian

Reply:
sure, but that's not visible here. here it's a bit surprising to have this lone definition of a byteswapping helper that hardcodes this.

Reply:
Please indent the nested defines by two spaces, `#  define`, `#  error`, etc.